src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorRequestContextFactory.cs

```diff
@@ -22,7 +24,8 @@ public RazorRequestContextFactory(ILspServices lspServices)
 
     public Task<RazorRequestContext> CreateRequestContextAsync<TRequestParams>(IQueueItem<RazorRequestContext> queueItem, TRequestParams @params, CancellationToken cancellationToken)
     {
-        var logger = _lspServices.GetRequiredService<LoggerAdapter>();
+        // To make things easier for endpoints, we create a logger here for them, with the LSP message as the category
+        var logger = _lspServices.GetRequiredService<IRazorLoggerFactory>().CreateLogger(queueItem.MethodName);
```

Comment:
The other option here we could do as a follow up is to just remove the `Logger` property from `RequestContext` altogether, and just have the endpoints take an `IRazorLoggerFactory` and call `CreateLogger` just like everything else.

Reply:
Yes would be good to remove Logger from RequestContext if possible

Reply:
Always happy to open more PRs. This one is probably big enough :)

Reply:
https://github.com/dotnet/razor/pull/9751